middleware/cors.js

```diff
@@ -0,0 +1,75 @@
+const cspConfig = require('../config/http-headers').content_security_policy;
+const accessControlConfig = require('../config/http-headers').access_control_allow_origin;
+const logger = require('../helpers/logger');
+
+if (!process.env.CORS) 	{
+	logger.info('CORS env has not been defined, to enable route specific CORS'
+	+ ' header set value to 1 and update values in config.cors');
+}
+
+
+const cspHeadersForRoute = (path, regexs, corsDefault) => {
+	let { defaultSrc, scriptSrc, objectSrc } = corsDefault;
+
+	const matchingKeys = Object.keys(regexs)
+		.filter(key => path.match(key));
+	const corsHeaders = matchingKeys.map(key => regexs[key]);
+	corsHeaders.forEach((matchingHeader) => {
+		if (matchingHeader.defaultSrc && matchingHeader.defaultSrc.includes('*')) {
+			defaultSrc = '*';
+		} else if (matchingHeader.defaultSrc) {
+			defaultSrc = `${defaultSrc} ${matchingHeader.defaultSrc}`;
+		}
+		if (matchingHeader.scriptSrc && matchingHeader.scriptSrc.includes('*')) {
+			scriptSrc = '*';
+		} else if (matchingHeader.scriptSrc) {
+			scriptSrc = `${scriptSrc} ${matchingHeader.scriptSrc}`;
+		}
+		if (matchingHeader.objectSrc && matchingHeader.objectSrc.includes('*')) {
+			objectSrc = '*';
+		} else if (matchingHeader.objectSrc) {
+			objectSrc = `${objectSrc} ${matchingHeader.objectSrc}`;
+		}
+	});
+	const finalCors = { defaultSrc, scriptSrc, objectSrc };
+	logger.debug(`cors headers for route ${path}`, finalCors);
+	return finalCors;
+};
+
+const accessControlHeadersForRoute = (path, regexs) => {
+	const matchingKeys = Object.keys(regexs)
```

Comment:
this duplication should be resolved

Reply:
the logic is the same except for 3 lines with cspHeadersForRoute. Makes sense to have a new function for it.